lua/rocks/api.lua

```diff
@@ -0,0 +1,110 @@
+---@mod rocks.api rocks.nvim Lua API
+---
+---@brief [[
+---
+---The Lua API for rocks.nvim.
+---Intended for use by modules that extend this plugin.
+---
+---@brief ]]
+
+-- Copyright (C) 2023 Neorocks Org.
+--
+-- Version:    0.1.0
+-- License:    GPLv3
+-- Created:    07 Dec 2023
+-- Updated:    07 Dec 2023
+-- Homepage:   https://github.com/nvim-neorocks/rocks.nvim
+-- Maintainer: NTBBloodbath <bloodbathalchemist@protonmail.com>
+
+---@class Rock
+---@field name string
+---@field version string
+
+---@alias rock_name string
+
+local api = {}
+
+local nio = require("nio")
+local cache = require("rocks.cache")
+local luarocks = require("rocks.luarocks")
+local fzy = require("rocks.fzy")
+local state = require("rocks.state")
+local commands = require("rocks.commands")
+local config = require("rocks.config.internal")
+
+---Tries to get the cached rocks.
+---Returns an empty list if the cache has not been populated
+---or no connection to luarocks.org can be established.
+---Will spawn an async task to attempt to populate the cache
+---if it is not ready.
+---@return table<rock_name, Rock[]> rocks
+function api.try_get_luarocks_rocks()
```

Comment:
same change as in the documentation :)

```suggestion
function api.try_get_cached_rocks()
```